avm/ptn/network/hub-networking/main.bicep

```diff
@@ -0,0 +1,501 @@
+      // Non-required parameters
+      diagnosticSettings: hub.value.diagnosticSettings ?? []
+      disableCopyPaste: hub.value.bastionHost.disableCopyPaste ?? true
+      enableFileCopy: hub.value.bastionHost.enableFileCopy ?? false
+      enableIpConnect: hub.value.bastionHost.enableIpConnect ?? false
+      enableShareableLink: hub.value.bastionHost.enableShareableLink ?? false
+      location: hub.value.location ?? location
+      enableTelemetry: hub.value.enableTelemetry ?? true
+      roleAssignments: hub.value.roleAssignments ?? []
+      scaleUnits: hub.value.bastionHost.scaleUnits ?? 4
+      skuName: hub.value.bastionHost.skuName ?? 'Standard'
+      tags: hub.value.tags ?? {}
+    }
+    dependsOn: hubVirtualNetwork
+  }
+]
+
+// Create Azure Firewall if enabled
+// AzureFirewallSubnet is required to deploy Azure Firewall service. This subnet must exist in the subnets array if you enable Azure Firewall.
+module hubAzureFirewall 'br/public:avm/res/network/azure-firewall:0.3.0' = [
+  for (hub, index) in items(hubVirtualNetworks ?? {}): if (hub.value.enableAzureFirewall) {
+    name: '${uniqueString(deployment().name, location)}-${hub.key}-naf'
+    params: {
+      // Required parameters
+      name: hub.key
+      // Conditional parameters
+      hubIPAddresses: hub.value.azureFirewallSettings.hubIpAddresses ?? {}
+      virtualHubId: hub.value.azureFirewallSettings.virtualHub ?? ''
+      virtualNetworkResourceId: hubVirtualNetwork[index].outputs.resourceId ?? ''
+      // Non-required parameters
+      additionalPublicIpConfigurations: hub.value.azureFirewallSettings.additionalPublicIpConfigurations ?? []
+      applicationRuleCollections: hub.value.azureFirewallSettings.applicationRuleCollections ?? []
+      azureSkuTier: hub.value.azureFirewallSettings.azureSkuTier ?? {}
+      diagnosticSettings: hub.value.diagnosticSettings ?? []
+      enableTelemetry: hub.value.enableTelemetry ?? true
+      firewallPolicyId: hub.value.azureFirewallSettings.firewallPolicyId ?? ''
+      location: hub.value.location ?? location
+      lock: hub.value.lock ?? {}
+      managementIPAddressObject: hub.value.azureFirewallSettings.managementIPAddressObject ?? ''
+      managementIPResourceID: hub.value.azureFirewallSettings.managementIPResourceID ?? ''
+      natRuleCollections: hub.value.azureFirewallSettings.natRuleCollections ?? []
+      networkRuleCollections: hub.value.azureFirewallSettings.networkRuleCollections ?? []
+      publicIPAddressObject: hub.value.azureFirewallSettings.publicIPAddressObject ?? {}
+      publicIPResourceID: hub.value.azureFirewallSettings.publicIPResourceID ?? ''
+      roleAssignments: hub.value.roleAssignments ?? []
+      tags: hub.value.tags ?? {}
+      threatIntelMode: hub.value.azureFirewallSettings.threatIntelMode ?? ''
+      zones: hub.value.azureFirewallSettings.zones ?? []
+    }
+    dependsOn: hubVirtualNetwork
+  }
+]
+
+module hubAzureFirewallSubnet 'modules/getSubnet.bicep' = [
```

Comment:
Please not that the VNET module deployment also returns an array of subnets. In other words, you may be able to avoid running this extra deployment by instead accessing this output like `hubVirtualNetwork.outputs.subnets`